test/headers.js

```diff
@@ -178,3 +178,28 @@ test('non-existent headers set to undefined are omitted', async t => {
 	const headers = JSON.parse(body);
 	t.false(Reflect.has(headers, 'blah'));
 });
+
+test('preserve port in host header if non-standard port', async t => {
+	const {body} = await got(s.url, {json: true});
+	t.is(body.host, 'localhost:' + s.port);
+});
+
+test('strip port in host header if explicit standard port (:80) & protocol (HTTP)', async t => {
+	const {body} = await got('http://httpbin.org:80/headers', {json: true});
```

Comment:
I don't think we should use remote server when we can just use our local server. Using a remote endpoint slows down the tests and makes them more fragile.

Reply:
Right... We could have used `nock`, but I guess this is good enough then.

Reply:
Wait, I forgot about `nock`. I'll just push a commit.